### PR TITLE
feat: add basic devcontainer support

### PR DESCRIPTION
**Pull request by jeluard**
**Motivation**

Ease setup for environment supporting [devcontainer](https://code.visualstudio.com/docs/devcontainers/containers).
e.g. allows MacOS users to easily develop with vscode under linux via [docker](https://code.visualstudio.com/docs/devcontainers/containers) or remote via [ssh](https://code.visualstudio.com/docs/remote/ssh). Also improves the [Codespaces](https://github.com/features/codespaces) experience.

### CODE REVIEW

**Review by wemeetagain — DISMISSED** [review]
LGTM

Have you tried using this?

**Review by nflaig — COMMENTED** [review]
I haven't tested devcontainers myself but is this VS code specific and requires an extension? Might be worth to document how to use this in contribution docs

**Comment by jeluard** [comment]
@wemeetagain yes it does work fine, you can test it in codespaces directly from my [fork](https://github.com/jeluard/lodestar/tree/jeluard/devcontainer). Great for PR review!
Capture d’écran 2024-04-17 à 22 52 14

Capture d’écran 2024-04-17 à 23 11 22


@nflaig good point, need some docs

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -47,6 +47,10 @@ Contributing to tests:
   - Do not pull unpinned versions from DockerHub (use deterministic tag) or Github (checkout commit not branch).
   - Carefully design tests that depend on timing sensitive events like p2p network e2e tests. Consider that Github runners are significantly less powerful than your development environment.
 
+## Devcontainer
+
+A [devcontainer](https://containers.dev/) [configuration](.devcontainer/devcontainer.json) is provided to help speed up linux based development environment setup. It will be used by [GitHub Codespaces](https://github.com/features/codespaces) or directly inside VSCode via your local through this [extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers).
```
Comment (nflaig) [inline CONTRIBUTING.md]: Does it still complain about spelling if you write `VS Code` ?

Comment (jeluard) [reply CONTRIBUTING.md]: It doesn't, good suggestion!

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.18.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.18.0) :tada:

**Pull request merged**